operator/src/main/java/dev/responsive/k8s/crd/kafkastreams/ScaleUpStrategySpec.java

```diff
@@ -0,0 +1,45 @@
+package dev.responsive.k8s.crd.kafkastreams;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class ScaleUpStrategySpec {
+  public enum Type {
```

Comment:
do we need some kind of default enum for UNKNOWN for forwards compatibility?

Reply:
This gets run when the policy is read from k8s and loaded into the operator. So we'd hit this case if a user upgrades, specifies a policy that uses some new scaling strategy, and then downgrades operator. In this case we can either have the operator reject the policy and then the user can re-issue it, or choose some default scaling behavior. I'd rather just keep things simple and do the former for now.